lib/ping.js

```diff
@@ -19,9 +21,51 @@ const pingRepository = (urls) => {
 }
 
 const checkDownloadSpeed = (packages) => {
+  const speedTest = (url, size) => {
+    const getProtocol = url => (new URL(url)).protocol === 'http:' ? http : https
+    let startTime
+    let protocol = getProtocol(url)
+
+    return new Promise((resolve, _) => {
+      return protocol.get(url, response => {
+        response.once('data', () => {
+          startTime = process.hrtime()
+        });
+
+        response.once('end', () => {
+          const endTime = process.hrtime(startTime)
+          const duration = (endTime[0] * 1e9 + endTime[1]) / 1e9
+          const bitsLoaded = size * 8
+          const bps = (bitsLoaded / duration).toFixed(2)
+          const kbps = (bps / 1000).toFixed(2)
+          const mbps = (kbps / 1000).toFixed(2)
+          resolve({ bps, kbps, mbps })
+        })
+      })
+    }).catch(error => {
+      throw new Error(error);
+    })
+  }
+
   const promises = _.map(elem => {
-    return testConnection.checkDownloadSpeed(elem.url, elem.size)
-      .then(speed => ({ url: elem.url, speed: speed.mbps * 0.125 }))
+    const sample = []
+    for (let i = 0; i < DOWNLOAD_SAMPLE_SIZE; i++) {
+      sample.push(speedTest)
+    }
```

Comment:
This should be equivalent no?
```suggestion
    const sample = _.times(speedTest, DOWNLOAD_SAMPLE_SIZE)
```